netdata/netdata
# Better memory allocation for pages in the cache

##### Summary
When pages are added in the page cache the agent will currently allocate the entire page size (currently 4096 bytes). For partially filled pages (ie. when you frequently restart the agent) this will waste a lot of memory. 

This also makes the page cache inefficient since the amount of pages that will fit the cache are pre calculated based on the fixed page size. 

This PR will do the following:
- Allocate as much memory as needed (with a small rounding) when fetching pages from disk (stored metrics)
- Keep track of the allocated memory for the page cache (as pages are allocated and freed)
- Accommodate more pages in cache based on the actual memory usage and not page count 
- Pages for collected metrics are still pre allocated to their maximum page size (4096 bytes currently)
- To take advantage of this feature you must use `dbengine page cache with malloc = yes`

##### Test Plan
- Start the agent with this PR (set `dbengine page cache with malloc = yes` in the netdata.conf)
   - Check  `Netdata DB engine RAM usage (netdata.dbengine_ram)` under `Netdata Monitoring`
